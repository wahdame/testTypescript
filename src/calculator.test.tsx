import { render, waitFor, screen } from "@testing-library/react";
import axios from "axios";
import Todos from "./calculator";

jest.mock("axios");

const dummyTodos = [
  {
    userId: 1,
    id: 1,
    title: "todo 1",
    completed: false,
  },
  {
    userId: 1,
    id: 2,
    title: "todo 2",
    completed: false,
  },
  {
    userId: 1,
    id: 3,
    title: "todo 3",
    completed: false,
  },
];

test("todos list", async () => {
  //   axios.get = jest.fn().mockResolvedValue({ data: dummyTodos });
  (axios.get as jest.Mock).mockResolvedValue({ data: dummyTodos });

  // axios.get.mockResolvedValue({ data: dummyTodos });
  render(<Todos />);

  const todoList = await waitFor(() => screen.findAllByTestId("todo"));

  expect(todoList).toHaveLength(3);
});

test("todos data", async () => {
  //   axios.get = jest.fn().mockResolvedValue({ data: dummyTodos });

  (axios.get as jest.Mock).mockResolvedValue({ data: dummyTodos });
  render(<Todos />);

  const todoList = await waitFor(() => screen.findAllByTestId("todo"));

  expect(todoList).toMatchSnapshot('<li data-testid="todo">todo 2</li>');
});

test("todos object", async () => {
  const elements = { id: 1, name: "A" }; // should throw assertion error

  expect(elements).toHaveProperty("name");
});

test("a must be either number or String always", () => {
  let response = [{ a: "asdasd", y: 2 }, { a: true }];
  expect(response).toEqual(
    expect.arrayContaining([
      expect.objectContaining({
        a: expect.any(String) || expect.any(Boolean),
        y: expect.any(Number) || expect.any(null),
      }),
    ])
  );
});

test("should contain important value in object", () => {
  const object = {
    important: "important",
    ignore: "ignore",
  };

  expect(object).toEqual(
    expect.objectContaining({
      important: "important",
    })
  );
});
