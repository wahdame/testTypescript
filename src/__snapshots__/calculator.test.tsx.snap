// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos data: <li data-testid="todo">todo 2</li> 1`] = `
Array [
  <li
    data-testid="todo"
  >
    todo 1
  </li>,
  <li
    data-testid="todo"
  >
    todo 2
  </li>,
  <li
    data-testid="todo"
  >
    todo 3
  </li>,
]
`;
